src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -32,6 +33,22 @@
 
   private DatastoreService datastore;
 
+  /** 
+   *  Checks if a user was already stored, done when a user logs in
+   * 
+   *  @return 0 on success, -1 when user was not found
+   */
+  private int userFound(int hash){
+    Query query =
+        new Query("User")
+            .setFilter(new Query.FilterPredicate("hashcode", FilterOperator.EQUAL, hash));
+    PreparedQuery results = datastore.prepare(query);
+    if (results.asSingleEntity() == null) {
```

Comment:
I'd suggest doing this as 
return results.asSingleEntity() == null ? -1 : 0;
However, I think returning ints is not ideal here anyway.

Reply:
I wasn't sure how ternary operators would be viewed so played it a bit safe